### DISCUSSION TITLE
VITS increases the rate of speech for longer text

### DISCUSSION THREAD

**Discussion by nicemanis** [coqui-ai/TTS#3630] [General Q&A]
I've noticed that my VITS model tends to speed up the synthesized speech as the input text gets longer. Here is a plot of how characters per second changes as the number of input characters are increased:
![image](https://github.com/coqui-ai/TTS/assets/9463816/c4c7fc97-c61a-444a-9c36-afb99bbb1687)

The same plot for training data looks like this:
![image](https://github.com/coqui-ai/TTS/assets/9463816/3f17c2b1-3725-4e16-952b-41603c69db8a)

The rate of speech should be +/- the same, regardless of the input text length.
Has anyone experienced this issue or have any ideas what causes it or how to solve it?

**Comment by eginhard** [comment]
It's hard to draw conclusions from these plots because the scales are so different but I think they show relatively similar behaviour. Could you show both plots with the same axis limits and only showing y-axis from 0 to ~30? Also make sure to trim any silence from the data.

**Reply by nicemanis** [reply]
Here is my VITS model synthesized audio speech rates:
![image](https://github.com/coqui-ai/TTS/assets/9463816/cb4dbc93-176a-4b1e-8b50-f72b0a1ac557)

Heere are the training data speech rates, silences are trimed at the beginning/end of audio, silences between words are still present:
![image](https://github.com/coqui-ai/TTS/assets/9463816/1525db96-bf57-4ba1-9eef-1d5034396b08)


Anyhow, I think the main hypothesis "VITS increases rate of speech for longer text" is true. The main question is - why there is such behaviour and how to fix it. My guess is that SDP is at fault.

**Reply by eginhard** [reply]
Did you also trim silence from the TTS output?

**Reply by nicemanis** [reply]
I did not, but the fact that the speech speeds up is very obvious when listening to samples.
